### PR TITLE
K8SPSMDB-342: make healthcheck to use certificates

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPSMDB-342](https://jira.percona.com/browse/K8SPSMDB-342)

Provide ssl flags to healthcheck to use tls certificates when requesting mongo

### CODE REVIEW

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 1]
```diff
@@ -153,6 +153,10 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 							"/data/db/mongodb-healthcheck",
 							"k8s", "liveness",
 							"--component", "mongos",
+							"--ssl",
+							"--sslCAFile", "/etc/mongodb-ssl/ca.crt",
+							"--sslPEMKeyFile", "/tmp/tls.pem",
+							"--sslInsecure",
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:159]: it should be added only if Operator >= 1.7.0

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/apis/psmdb/v1/psmdb_defaults.go (hunk 1 above)
Comment (hors) [reply pkg/apis/psmdb/v1/psmdb_defaults.go:159]: @delgod I think you mean >= 1.6.0

Comment (qJkee) [reply pkg/apis/psmdb/v1/psmdb_defaults.go]: 798d54a

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| arbiter | passed |
| service-per-pod | failed |
| liveness | passed |
| smart-update | failed |
| version-service | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| init-deploy | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| demand-backup-sharded | failed |
| rs-shard-migration | failed |
| arbiter | passed |
| service-per-pod | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| liveness | passed |
| smart-update | passed |
| version-service | failed |

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 2]
```diff
@@ -153,11 +153,17 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 							"/data/db/mongodb-healthcheck",
 							"k8s", "liveness",
 							"--component", "mongos",
+							"--ssl",
+							"--sslInsecure",
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:157]: can we add these options only if Operator >= 1.7.0

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| service-per-pod | failed |
| upgrade-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 3]
```diff
@@ -157,7 +157,12 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 					},
 				},
 			}
-
+			if cr.CompareVersion("1.6.0") >= 0 {
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:160]: please 1.7.0

**Review by delgod — CHANGES_REQUESTED** [review]

File: e2e-tests/arbiter/compare/statefulset_arbiter-rs0-arbiter.yml [hunk 4]
```diff
@@ -87,6 +87,12 @@ spec:
                 - /data/db/mongodb-healthcheck
                 - k8s
                 - liveness
+                - --ssl
+                - --sslInsecure
+                - --sslCAFile
+                - /etc/mongodb-ssl/ca.crt
+                - --sslPEMKeyFile
+                - /tmp/tls.pem
```
Comment (blez) [inline e2e-tests/arbiter/compare/statefulset_arbiter-rs0-arbiter.yml:95]: This files depend on secret existence.
But secret not always can exists. And if is doesn't, readiness and liveness probe will fail.

File: pkg/apis/psmdb/v1/psmdb_defaults.go (hunk 3 above)
Comment (qJkee) [reply pkg/apis/psmdb/v1/psmdb_defaults.go]: fixed as a part of d7755107

File: e2e-tests/arbiter/compare/statefulset_arbiter-rs0-arbiter.yml (hunk 4 above)
Comment (qJkee) [reply e2e-tests/arbiter/compare/statefulset_arbiter-rs0-arbiter.yml:95]: fixed as a part of d7755107

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | failed |
| one-pod | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | failed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| demand-backup-sharded | passed |
| users | passed |
| upgrade | failed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| storage | passed |
| init-deploy | passed |
| self-healing | passed |
| limits | passed |
| operator-self-healing | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | failed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | failed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | failed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | failed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| self-healing | passed |
| limits | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade-consistency | failed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | failed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-509-4df645b2

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| operator-self-healing | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| scheduled-backup | passed |
| users | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| scheduled-backup | failed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| demand-backup-sharded | passed |
| upgrade | failed |
| upgrade-sharded | passed |

**Review by blez — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**